File: Backend/Services/OrderService/OrderService.Application/DTOs/Order/OrderDto.cs
using OrderService.Application.DTOs.Meal;

namespace OrderService.Application.DTOs.Order
{
    public class OrderDto
    {
        public Guid Id { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public Guid ClientId { get; set; }
        public Guid? CourierId { get; set; }
        public string Status { get; set; }
        public string Address { get; set; }
        public List<OrderMealDto> Meals { get; set; } = new();
        public decimal TotalPrice { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? DeliveryDate { get; set; }
    }
}

File: Frontend/angular-client/src/app/models/order/order-dto.ts
import { MealInOrder } from "../meal/meal-in-order";
import { OrderStatus } from "./order-status";

export interface OrderDto{
    id: string;
    orderNumber: string;
    clientId: string;
    courierId: string | null;
    status: OrderStatus;
    address: string;
    meals: MealInOrder[];
    totalPrice: number;
    createdAt: string;
    confirmedByClient: boolean;
    confirmedByCourier: boolean;
    deliveryDate: string | null;
}